File: Typescript.js/RecentMistakesFirstSorter.ts
import { Card } from "./Card";
import { CardOrganizer } from "./CardOrganizer";

export class RecentMistakesFirstSorter implements CardOrganizer {
  public organize(cards: Card[]): Card[] {
    const incorrectCards: Card[] = [];
    const correctCards: Card[] = [];

    for (const c of cards) {
      if (c.wasCorrectInLastRound()) {
        correctCards.push(c);
      } else {
        incorrectCards.push(c);
      }
    }

    const reusalt: Card[] = [];
    reusalt.push(...incorrectCards);
    reusalt.push(...correctCards);

    return reusalt;
  }
}

File: Java/RecentMistakesFirstSorter.java
package Java;
import java.util.ArrayList;
import java.util.List;

public class RecentMistakesFirstSorter implements CardOrganizer {

    @Override
    public List<Card> organize(List<Card> cards) {

        List<Card> incorrect = new ArrayList<>();
        List<Card> correct = new ArrayList<>();

        for (Card c : cards) {
            if (c.wascorrectInLastRound()) {
                correct.add(c);
            } else {
                incorrect.add(c);
            }
        }

        List<Card> result = new ArrayList<>();
        result.addAll(incorrect);
        result.addAll(correct);

        return result;
    }
}

    


